### PR TITLE
typo in gdscript_workspace.cpp fixed

### LINKED ISSUE #36389: Strange line(copy paste bug?) in gdscript_workspace.cpp
**Godot version:**
Godot 4.0 5a8107f70a6dc3c43e89

**Issue description:**

This line looks very, very strange.
`script->get()` always return same value

https://github.com/godotengine/godot/blob/5a8107f70a6dc3c43e89df0ad69a5a0124bfc2df/modules/gdscript/language_server/gdscript_workspace.cpp#L52-L54

### PR DESCRIPTION
**Pull request by ThakeeNathees**
Fix: #36389

### CODE REVIEW

File: modules/gdscript/language_server/gdscript_workspace.cpp [hunk 1]
```diff
@@ -53,7 +53,7 @@ void GDScriptWorkspace::remove_cache_parser(const String &p_path) {
 	Map<String, ExtendGDScriptParser *>::Element *parser = parse_results.find(p_path);
 	Map<String, ExtendGDScriptParser *>::Element *script = scripts.find(p_path);
 	if (parser && script) {
-		if (script->get() && script->get() == script->get()) {
+		if (script->get() == parser->get()) {
```
Comment (akien-mga) [inline modules/gdscript/language_server/gdscript_workspace.cpp]: Maybe the whole block of lines 55-70 could be simplified further? @Geequlim 
```cpp
if (parser) {
	memdelete(parser->get());
	parse_results.erase(p_path);
}
if (script) {
	if (script->get()) { // May have been freed already if same as parser.
		memdelete(script->get());
	}
	scripts.erase(p_path);
}
```

Comment (ThakeeNathees) [reply modules/gdscript/language_server/gdscript_workspace.cpp]: yeah looks better, On it

Comment (ThakeeNathees) [reply modules/gdscript/language_server/gdscript_workspace.cpp]:
```
if (parser && parser->get()) {
	memdelete(parser->get());
	parse_results.erase(p_path);
}
if (script && script->get()) {
	memdelete(script->get());
	scripts.erase(p_path);
}
```

Comment (akien-mga) [reply modules/gdscript/language_server/gdscript_workspace.cpp]: I don't know, the previous logic would remove the path from `parse_results` and `scripts` even if the `ExtendGDScriptParser` instance is null, at least in some branches.

**Comment by ThakeeNathees** [comment]
the method `parse_results.erase(p_path);` tries to find `p_path` element inside by `find(p_path)` internally and if the there is a possibility to occur any error due to the `p_path` is `NULL` then the first line must also have to be handled ( not sure about it )
`Map<String, ExtendGDScriptParser *>::Element *parser = parse_results.find(p_path);`
@akien-mga

File: modules/gdscript/language_server/gdscript_workspace.cpp (hunk 1 above)
Comment (ThakeeNathees) [reply modules/gdscript/language_server/gdscript_workspace.cpp]:
```c++
if (parser) {
	if (parser->get()) memdelete(parser->get());
	parse_results.erase(p_path);
}
if (script) {
	if (script->get()) memdelete(script->get());
	scripts.erase(p_path);
}
```

File: modules/gdscript/language_server/gdscript_workspace.cpp [hunk 2]
```diff
@@ -52,20 +52,13 @@ void GDScriptWorkspace::_bind_methods() {
 void GDScriptWorkspace::remove_cache_parser(const String &p_path) {
 	Map<String, ExtendGDScriptParser *>::Element *parser = parse_results.find(p_path);
 	Map<String, ExtendGDScriptParser *>::Element *script = scripts.find(p_path);
-	if (parser && script) {
-		if (script->get() && script->get() == script->get()) {
```
Comment (Geequlim) [inline modules/gdscript/language_server/gdscript_workspace.cpp]: Change to 
```
if (script->get() && script->get() == parser ->get()) {
```
should be ok.

Comment (ThakeeNathees) [reply modules/gdscript/language_server/gdscript_workspace.cpp]: Done.
and thanks for the review

Comment (akien-mga) [reply modules/gdscript/language_server/gdscript_workspace.cpp]: @Geequlim See the previous discussion though, there's a lot of redundant code in 
https://github.com/godotengine/godot/blob/07c99340d1253799675cdf341c633c29ad0164ad/modules/gdscript/language_server/gdscript_workspace.cpp#L55-L70

**Comment by Geequlim** [comment]
If you really need less condition checks you can using a long checking like this
```c++
if (parser && parser->get()) memdelete(parser->get());
if ((script && script->get()) && ((parser && parser->get() != script->get()) || !parser)) memdelete(script->get());
parse_results.erase(p_path);
scripts.erase(p_path);
```

**Comment by akien-mga** [comment]
I think you're missing the point of how this can be simplified, but let's just go with the simple typo fix we have now.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.1.

**Pull request merged**